### PR TITLE
List Blobs Versioning/Prefix/Delimiter support 

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Adding test to verify List Blobs Versioning/Prefix/Delimiter support 

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/container/client_test.go [hunk 1]
```diff
@@ -748,6 +748,45 @@ func (s *ContainerRecordedTestsSuite) TestContainerListBlobsInvalidDelimiter() {
...
+	containerClient := testcommon.CreateNewContainer(context.Background(), _require, containerName, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	blobNames := []string{"a/1", "a/2", "b/1", "blob"}
+	blobMap := testcommon.BlobListToMap(blobNames)
+	for _, name := range blobNames {
+		testcommon.CreateNewBlockBlob(context.Background(), _require, name, containerClient)
+	}
+
+	opts := container.ListBlobsHierarchyOptions{
```
Comment (seanmcc-msft) [inline sdk/storage/azblob/container/client_test.go:767]: We also need to include the Delimiter.  See the .NET PR - https://github.com/Azure/azure-sdk-for-net/pull/30247/files#diff-cfbbd86d890e76d951dd6142c352ee932045fb0ce70ceb31c1d069f404c692c2R2929

Comment (seanmcc-msft) [reply sdk/storage/azblob/container/client_test.go:767]: See the REST docs for more details about what the delimiter is - https://learn.microsoft.com/en-us/rest/api/storageservices/list-blobs?tabs=azure-ad#uri-parameters

Comment (siminsavani-msft) [reply sdk/storage/azblob/container/client_test.go:767]: Delimiter for NewListBlobsHierarchyPager is the first parameter of the method (as seen in line  774). It is not listed as an option for ListBlobsHierarchyOptions. Is this something we should reconsider?

Comment (seanmcc-msft) [reply sdk/storage/azblob/container/client_test.go:767]: Nope, this looks good, I missed that.

**Review by seanmcc-msft — APPROVED** [review]

**Review by gapra-msft — APPROVED** [review]

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**